bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BTable.java

```diff
@@ -117,6 +115,11 @@ public BType getType() {
 
 
     public boolean hasNext(boolean isInTransaction) {
+        if (this.isInMemoryTable) {
+            if (this.iterator == null) {
```

Comment:
shall we merge these two conditions in to one if clause?

Reply:
Fixed. Thanks .